DashAI/front/src/components/ConfigurableObject/FormInputs.jsx

```diff
@@ -0,0 +1,377 @@
+      <IconButton color="primary" component="label" onClick={handleButtonClick}>
+        <SettingsIcon />
+      </IconButton>
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>{`${selectedOption} parameters`}</DialogTitle>
+        <DialogContent key={selectedOption}>
+          {defaultValues.loaded && (
+            <Subform
+              name={name}
+              parameterSchema={paramSchema}
+              setFieldValue={setFieldValue}
+              choice={selectedOption}
+              defaultValues={defaultValues.values}
+            />
+          )}
+        </DialogContent>
+        <DialogActions>
+          <Button variant="outlined" onClick={handleClose} autoFocus>
+            Save
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </div>
+  );
+}
+
+ClassInput.propTypes = {
+  name: PropTypes.string.isRequired,
+  paramJsonSchema: PropTypes.objectOf(
+    PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.object])
+  ).isRequired,
+  setFieldValue: PropTypes.func.isRequired,
+  formDefaultValues: PropTypes.objectOf(
+    PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool,
+      PropTypes.number,
+      PropTypes.object,
+    ])
+  ),
+};
+
+ClassInput.defaultProps = {
+  formDefaultValues: { emptyDefaultValues: true },
+};
+
+export function genInput(objName, paramJsonSchema, formik, defaultValues) {
+  const { type, properties } = paramJsonSchema;
+  // Props that are common to almost all form inputs.
+  const commonProps = {
+    name: objName,
+    value: formik.values[objName],
+    onChange: formik.handleChange,
+    error: formik.errors[objName],
+    description: paramJsonSchema.description,
+    key: objName,
+  };
+  switch (type) {
+    case "object":
+      return (
+        <Stack key={objName} spacing={3}>
+          {Object.keys(properties).map((parameter) =>
+            genInput(
+              parameter,
+              properties[parameter].oneOf[0],
+              formik,
+              defaultValues[parameter]
+            )
+          )}
+        </Stack>
+      );
+    case "class":
+      return (
+        <ClassInput
+          name={objName}
+          paramJsonSchema={paramJsonSchema}
+          setFieldValue={formik.setFieldValue}
+          formDefaultValues={defaultValues}
+          key={`rec-param-${objName}`}
+        />
+      );
+    case "integer":
+      return <Integer {...commonProps} />;
+    case "number":
+      return <Number {...commonProps} />;
+    case "string":
+      return <Select {...commonProps} options={paramJsonSchema.enum} />;
+    case "text":
+      return <Text {...commonProps} />;
+    case "boolean":
+      return <Boolean {...commonProps} />;
+    case "float":
+      return <Float {...commonProps} />;
+    default:
+      return (
+        <Typography variant="p" sx={{ color: "red" }}>
+          {`Not a valid parameter type ${type}`}
+        </Typography>
```

Comment:
I think the correct thing to do in this case is for the code to throw an exception.

```suggestion
      throw new Error(`Error while rendering ${objName}: ${type} is not a valid parameter type.`)
```